plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/utils/UIUtils.java

```diff
@@ -485,7 +485,10 @@ public static ElementTreeSelectionDialog createFileDialog(String selectedFile,
             @Override
             public boolean select(Viewer viewer, Object parentElement, Object element) {
                 return element instanceof IContainer 
-                        || (element instanceof IFile && ((IFile)element).getFileExtension().equals(extension));
+                        || (element instanceof IFile 
+                        		&& extension != null 
```

Comment:
Currently, this method is only used to select 'json' files so that we are free to set the null case logic. What you suggest is ok to me. Please add it to Javadoc comments, too.